From 29dba221c6eb035ba476a195cb91fe9e1d5b983c Mon Sep 17 00:00:00 2001
From: Jon TURNEY <jon.turney@dronecode.org.uk>
Date: Wed, 21 Nov 2012 18:36:03 +0000
Subject: Fix compilation when ./configure'd with --disable-asm

In file included from glapi_dispatch.c:90:0:
../../../src/mapi/glapi/glapitemp.h:1640:45: error: no previous prototype for 'glReadBufferNV' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:4198:45: error: no previous prototype for 'glDrawBuffersNV' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6377:45: error: no previous prototype for 'glFlushMappedBufferRangeEXT' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6389:49: error: no previous prototype for 'glMapBufferRangeEXT' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6401:45: error: no previous prototype for 'glBindVertexArrayOES' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6413:45: error: no previous prototype for 'glDeleteVertexArraysOES' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6433:45: error: no previous prototype for 'glGenVertexArraysOES' [-Wmissing-prototypes]
../../../src/mapi/glapi/glapitemp.h:6445:50: error: no previous prototype for 'glIsVertexArrayOES' [-Wmissing-prototypes]

See also commits 8f3be339 (which fixed this once already), and 3ed8d42 (which
reverted the fix to fix it somewhere else [1])

[1] http://lists.freedesktop.org/archives/mesa-dev/2012-July/023899.html

Signed-off-by: Jon TURNEY <jon.turney@dronecode.org.uk>
---
diff --git a/src/mapi/glapi/glapi_dispatch.c b/src/mapi/glapi/glapi_dispatch.c
index be65ebe..e73ead5 100644
--- a/src/mapi/glapi/glapi_dispatch.c
+++ b/src/mapi/glapi/glapi_dispatch.c
@@ -87,6 +87,14 @@
 /* those link to libglapi.a should provide the entry points */
 #define _GLAPI_SKIP_PROTO_ENTRY_POINTS
 #endif
+
+#undef GL_COVERAGE_SAMPLES_NV
+#define __gl3_h_
+#define GL_EXT_multisampled_render_to_texture 1
+#define GL_EXT_texture_storage 1
+#include "GLES2/gl2platform.h"
+#include "GLES2/gl2ext.h"
+
 #include "glapi/glapitemp.h"
 
 #endif /* USE_X86_ASM */
--
cgit v0.9.0.2-2-gbebe
